std/range/package.d

```diff
@@ -10761,3 +10764,477 @@ pure unittest
     static immutable r2 = [1, 2, 3, 4, 0, 0];
     assert(r1.padRight(0, 6).equal(r2));
 }
+
+/**
+`orElse` provides a safe wrapper around possibly empty ranges.
+If an empty range is passed, a range of length 1 with the given fallback will
+be used. Otherwise, the original range is used for all range methods.
+Moreover, all range capabilities of the original range are forwared.
+
+Params:
+    r = input range (can be empty)
+    fallback = element to build a single-element range from if r is empty
+
+Returns: if the range is non-empty the original range and otherwise a range of length
+with the fallback.
+*/
+auto orElse(Range, RangeElementType = ElementType!Range)(Range r, RangeElementType fallback)
+if (isInputRange!(Unqual!(Range)) &&
+    !is(CommonType!(ElementType!Range, RangeElementType) == void))
+{
+    static struct OrElseRange
+    {
+        private
```

Comment:
I'm edging close to preferring individual `private` with declarations because:

* labels (`private:`) affect things at distance and make reading and reviewing more complicated
* scopes (like here) force an extra indent level
* refactoring/blame is by far best with individual labels
* not a matter of scaling (number of symbols is tractable)